src/Raven.Client/Documents/Session/IAdvancedDocumentSessionOperations.cs

```diff
@@ -170,7 +170,13 @@ public interface IAdvancedDocumentSessionOperations
         /// SaveChanges will wait for the indexes to catch up with the saved changes
         /// </summary>
         void WaitForIndexesAfterSaveChanges(TimeSpan? timeout = null, bool throwOnTimeout = true, string[] indexes = null);
-        
+
+        /// <summary>
+        /// Overwrite the existing transaction mode
+        /// </summary>
+        /// <param name="mode"></param>
+        void SetTransactionMode(TransactionMode mode);
```

Comment:
This is validated on `SaveChanges`, IIRC. So if we set it to normal but we have added cmpxchng, we'll throw.

Reply:
Will add a test to verify the behavior.